### PR TITLE
Tidy up

### PR DESCRIPTION
**Pull request by janm399**
Just a few improvements to the exercise filtering. The key concepts are:

* separate side-effects from pure computation
* favour immutability
* reason about optional values (namely is something _always_ ``.None`` / ``nil``?)

### CODE REVIEW

File: Muvr/MRLabelViewController.swift [hunk 1]
```diff
@@ -63,46 +63,31 @@ class MRLabelViewController : UIViewController, UITableViewDelegate, UITableView
...
-        let index = indexPath.row as Int
-        if cell == nil {
-            cell = UITableViewCell(style: UITableViewCellStyle.Value1, reuseIdentifier: autoCompleteRowIdentifier)
-        }
-        cell?.textLabel?.text = autocompleteExercises[index]
-        return cell!
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier("autocompleteRowIdentifier") as UITableViewCell!
+        cell.textLabel?.text = autocompleteExercises[indexPath.row]
+        return cell
```
Comment (samudurand) [inline Muvr/MRLabelViewController.swift]: Will it not crash if there is no cell yet ?

Comment (janm399) [inline Muvr/MRLabelViewController.swift]: There is one now in the IB / storyboard.

Comment (samudurand) [inline Muvr/MRLabelViewController.swift]: Too late

Comment (samudurand) [inline Muvr/MRLabelViewController.swift]: alright seems it works well like that

**Pull request merged**